rust/osm2lanes/src/transform/tags_to_lanes/road.rs

```diff
@@ -0,0 +1,412 @@
+use std::str::FromStr;
+
+use Highway::*;
+use Hunch::*;
+
+use crate::road::LaneDirection;
+use crate::tag::TagKey;
+use crate::transform::RoadMsg::Unimplemented;
+use crate::transform::{RoadFromTags, RoadMsg, RoadWarnings, Tags};
+use crate::Locale;
+
+pub fn tags_to_lanes(tags: &Tags, locale: &Locale) -> Result<RoadFromTags, RoadError> {
+    let mut road: Road = Road::new(locale);
+
+    road.set_tags::<Oneway>(tags);
+    road.set_tags::<Busway>(&tags);
+    road.set_tags::<Lanes>(&tags);
+
+    // if (guessing_mode)
+    {
+        road.guess::<Oneway>();
+        road.guess::<Busway>();
+        road.guess::<Lanes>();
+    }
+
+    road.calculate_lanes_ltr().into()
+}
+
+/// A collection of schemas. We have a Hunch about each schema because we need at least an option,
+/// so why not track it as a Hunch?
+struct Road {
+    busway: Hunch<Busway>,
+    lanes: Hunch<Lanes>,
```

Comment:
how does this work? do we make assumptions about how lanes vs busways are arranged?

Reply:
We already do, and we need to because people want to tag just `busway:both=lane` to mean 4 lanes: a default 2 normal lanes with a busway on the outside, which we currently support with test [`way_id: 988980354`](https://github.com/a-b-street/osm2lanes/blob/620a83332cd696e08b41ce89e1d559ce7e1a386f/data/tests.yml#L815). It's why `busway` chooses `forward_inside()` vs `forward_outside()`.